src/ChangeStream.php

```diff
@@ -55,14 +57,17 @@ public function getId()
 
     public function key()
     {
-
+        return $this->key;
     }
 
     public function next()
     {
+        $this->key++;
```

Comment:
Not every advancement with `next()` is going to be successful. This looks like it's possible for `$this->key` to get incremented more times than necessary.

I think a test where there is no next element (because no changes have occurred) or we end up resuming a few times would demonstrate that the key can get out of sync.